server/issue_test.go

```diff
@@ -150,5 +150,5 @@ func TestRouteIssueTransition(t *testing.T) {
 	request.Header.Set("Mattermost-User-Id", "connected_user")
 	w := httptest.NewRecorder()
 	p.ServeHTTP(&plugin.Context{}, w, request)
-	assert.Equal(t, 400, w.Result().StatusCode)
+	assert.Equal(t, 200, w.Result().StatusCode)
```

Comment:
@jfrerich Do you know why this value changed for this PR?

Reply:
honestly, no.  I noticed this comment in the PR that added the code.  https://github.com/mattermost/mattermost-plugin-jira/pull/454/files#r393127688

Reply:
I'll investigate what changed and get back

Reply:
This one I am still a bit confused.  `w` is a NewRecoder() which returns `w.Code` == `200`

`w.Result().StatusCode` `net/http/httptest` appears to be returning that value.

Reply:
@levb, @mickmister Here's the explanation for the change from 400 -> 200 status code.

@levb in your PR, you removed the `http.Error(w, err.Error(), status)` from the `ServerHttp` method.  This call writes the writes the `Code` and `HeaderMap` fields of the json `w` object.  (keep this in mind during the flow of the test)

`w := httptest.NewRecorder()`  ->  This initializes the `w` object to the following:
```json
w = {
     "Code": 200,
     "HeaderMap": {},
     "Body": {},
     "Flushed": false
}
```

`ServeHTTP` calls p.serveHTTP() and returns `400`

By removing `http.Error(w, err.Error(), status)` (the reminder mentioned first) w.Code no longer gets written and remains `200` instead of `400`

Does that explanation make sense?

Reply:
@jfrerich I think I understand. With your judgment, would you say this is which of the following:

- a bug in our code handling that request
- an incorrect assertion in the test
- expected behavior

Reply:
@mickmister, this actually was a bug in the code.  http.Error() should be called so the StatusCode is written by the http.ResponseWriter.

To fix this, I changed the returns to use the reposndErr() function which includes the http.Error() function call.

The test itself wasn't very thorough, so I updated it to add coverage for the httpAPITranstionIssue() function.